python/sparkdl/estimators/keras_image_file_estimator.py

```diff
@@ -0,0 +1,262 @@
+                  imageLoader=None, outputMode="vector", optimizer=None, loss=None,
+                  kerasParams=None):
+        """
+        kwargs = self._input_kwargs
+        self._set(**kwargs)
+        return self
+
+    def setLabelCol(self, value):
+        return self._set(labelCol=value)
+    
+    def getLabelCol(self):
+        return self.getOrDefault(self.labelCol)
+
+    def setOptimizer(self, value):
+        return self._set(optimizer=value)
+
+    def getOptimizer(self):
+        return self.getOrDefault(self.optimizer)
+
+    def setLoss(self, value):
+        return self._set(loss=value)
+
+    def getLoss(self):
+        return self.getOrDefault(self.loss)
+    
+    def _validateParams(self):
+        """
+        Check Param values so we can throw errors on the driver, rather than workers.
+        :return: True if parameters are valid
+        """
+        if not self.isDefined(self.optimizer):
+            raise ValueError("Required Param optimizer must be specified but was not.")
+        if not is_valid_optimizer(self.getOptimizer()):
+            raise ValueError("Named optimizer not supported in Keras: %s" % self.getOptimizer())
+        if not self.isDefined(self.loss):
+            raise ValueError("Required Param loss must be specified but was not.")
+        if not is_valid_loss_function(self.getLoss()):
+            raise ValueError("Named loss not supported in Keras: %s" % self.getLoss())        
+        if not self.isDefined(self.inputCol):
+            raise ValueError("Input column must be defined")
+        if not self.isDefined(self.outputCol):
+            raise ValueError("Output column must be defined")
+        return True
+
+    def _localFit(self, featuresBc, labelsBc, modelBytesBc, paramMap):
+        # Copy the estimator to add to the Params without modifying this instance.        
+        print('[_localFit] paramMap', paramMap)
+        estimator = self.copy(paramMap)
+
+        features = featuresBc.value
+        labels = None if labelsBc is None else labelsBc.value
+        model = bytes_to_model(modelBytesBc.value)
+        
+        print('[_localFit] model name', model.name)
+
+        model.compile(optimizer=estimator.getOptimizer(), loss=estimator.getLoss())
+        kerasParams = estimator.getKerasParams()
+
+        print('[_localFit] Keras params', kerasParams)
+        print('[_localFit] feature', features.shape)
+
+        model.fit(x=features, y=labels, **kerasParams)
+        return model
+
+    def _getNumpyFeaturesAndLabels(self, dataset):
+        image_uri_col = self.getInputCol()
+        label_col = None
+        if self.isDefined(self.labelCol) and self.getLabelCol() != "":
+            label_col = self.getLabelCol()
+        tmp_image_col = self._loadedImageCol()
+        image_df = self._loadImages(dataset, image_uri_col).dropna(subset=[tmp_image_col])
+        localFeatures = list(map(
```

Comment:
This is an impressive 2-liner, but please use list comprehensions for readability. Note that you can distribute the array construct. It does not matter too much though as this is something we are probably going to move to the scala side.